File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/BannerPages/GrandMobilesView.swift
import SwiftUI

struct GrandMobilesView: View {
    @Environment(\.presentationMode) var presentationMode

    var body: some View {
        ZStack(alignment: .bottom) {
            // Background
            Color(red: 0.97, green: 0.98, blue: 0.99)  // background-light #F8FAFC
                .ignoresSafeArea()

            VStack(spacing: 0) {
                // Status Bar Placeholder (using standard safe area instead of hardcoded)

                // Header
                headerView

                // Scrollable Content
                ScrollView(showsIndicators: false) {
                    VStack(spacing: 0) {
                        // Hero Section
                        heroSectionView

                        // Shop by Brand
                        shopByBrandSection

                        // Trending Now
                        trendingNowSection

                        Spacer().frame(height: 120)
                    }
                }
            }

            // Fixed Bottom Nav
            bottomNavView
        }
        .navigationBarHidden(true)
        .ignoresSafeArea(.all, edges: .bottom)
    }

    // MARK: - Header
    private var headerView: some View {
        HStack {
            Button(action: {
                presentationMode.wrappedValue.dismiss()
            }) {
                Image(systemName: "chevron.left")
                    .font(.system(size: 20))
                    .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))  // primary #001B44
                    .padding(8)
                    .background(Color.white.opacity(0.2))
                    .clipShape(Circle())
            }

            Spacer()

            HStack(spacing: 12) {
                Button(action: {}) {
                    Image(systemName: "magnifyingglass")
                        .font(.system(size: 20))
                        .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                        .padding(8)
                        .background(Color.white.opacity(0.2))
                        .clipShape(Circle())
                }

                Button(action: {}) {
                    Image(systemName: "bag")
                        .font(.system(size: 20))
                        .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                        .padding(8)
                        .background(Color.white.opacity(0.2))
                        .clipShape(Circle())
                }
            }
        }
        .padding(.horizontal, 24)
        .padding(.vertical, 16)
    }

    // MARK: - Hero Section
    private var heroSectionView: some View {
        GeometryReader { geometry in
            ZStack {
                // Vibrant Gradient Background
                LinearGradient(
                    gradient: Gradient(colors: [
                        Color(red: 0.31, green: 0.69, blue: 0.68),  // #4FB0AE
                        Color(red: 0.83, green: 0.89, blue: 0.83),  // #D4E2D4
                        Color(red: 1.0, green: 0.84, blue: 0.44),  // #FFD670
                        Color(red: 1.0, green: 0.57, blue: 0.28),  // #FF9248
                    ]),
                    startPoint: .topLeading,
                    endPoint: .bottomTrailing
                )
                .overlay(
                    // Pattern Overlay Placeholder - mimicking the opacity: 0.15 texture
                    Color.black.opacity(0.05)
                )

                // Sparkles
                VStack {
                    HStack {
                        Image(systemName: "sparkles")
                            .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                            .font(.system(size: 24))
                            .padding(.top, 48)
                            .padding(.leading, 40)
                        Spacer()
                        Image(systemName: "sparkles")
                            .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27).opacity(0.4))
                            .font(.system(size: 24))
                            .padding(.top, 128)
                            .padding(.trailing, 48)
                    }
                    Spacer()
                    HStack {
                        Image(systemName: "sparkles")
                            .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                            .font(.system(size: 24))
                            .padding(.bottom, 160)  // Adjusted to not overlap phone too much
                            .padding(.leading, 64)
                        Spacer()
                    }
                }

                // Content
                VStack(spacing: 0) {
                    Text("Betul's Exclusive")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                        .tracking(-0.5)
                        .padding(.bottom, 8)

                    Text("GRAND MOBILES\nSALE")
                        .font(.system(size: 42, weight: .heavy))  // Plus Jakarta Sans style
                        .multilineTextAlignment(.center)
                        .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                        .lineSpacing(-10)  // Tight leading
                        .padding(.bottom, 32)

                    Button(action: {}) {
                        Text("GRAB NOW")
                            .font(.system(size: 18, weight: .black))
                            .tracking(2)
                            .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                            .padding(.horizontal, 40)
                            .padding(.vertical, 14)
                            .overlay(
                                RoundedRectangle(cornerRadius: 0)  // Square-ish styling from button
                                    .stroke(Color(red: 0.0, green: 0.11, blue: 0.27), lineWidth: 2)
                            )
                    }
                    .padding(.bottom, 48)

                    // Phones Row
                    HStack(alignment: .bottom, spacing: 4) {
                        phoneImage(
                            url:
                                "https://lh3.googleusercontent.com/aida-public/AB6AXuAxDDaHdTrwru2LB1m_bJMDHJJXXNMH6lz9-9S8rddKmEqEWunUOUFNIbDEZPU2d-YJNwhSmwIpL5fO0epqwuAxkKHsryedI3kKY1upVgvUY8zIeQO-cDyHSq6-D05NQA0G6nx_pJudqZF6b3iWA1wThCbVLqJxo19m7Zr0YUMv3dwjt7Sy6J4CKABFbQYaor4Ku67YWhfhByHnwYFOTfjsu2ML2U_ovNFWJfYxos7yBMXhlZSbcKqy41LlXGZAyT3a7GN-2zDoKY5S",
                            height: 128, yOffset: 16, totalWidth: geometry.size.width)
                        phoneImage(
                            url:
                                "https://lh3.googleusercontent.com/aida-public/AB6AXuCksjqMHl_Le2OSvtMgyqEeMo6F3kf3fFLKon7RNfxaxt4-VznnUV60f8NxHfim6HzcFbPp8g1xan7gXkkXG7QTymUM5a914tPRdFunL5GAdNouDtxDqqI6FpYeDg6JV-k3vMn3Vv59V5b21elRNYMVaMKilZnTQRoZyFd8qQTod4AQNKx_50Hsy9Guqv8as_l56QtYgg078rgL15j1EsNxw-sR6S-PfuyZuHLEfjxyMvpmJXWCuVnkuVWnVFPiHYLR7HYRSG6lytlr",
                            height: 160, yOffset: 8, totalWidth: geometry.size.width)
                        phoneImage(
                            url:
                                "https://lh3.googleusercontent.com/aida-public/AB6AXuC4DwQQFVyxxqGSPaN4wzRr5wT8_Pxg_2e8k2v_Yl0MZKth5yEZ0-GNsJqwmF9RPLRiCD-rjf7we_aMQjZpHUtp1TvWjCLfelyn9fmQDv_cr3EY2QCHDdBw5TOaVRYb7CGmoppynojkVeRqDBe1jHlv9sYjOsyksxwT0JUxoEZLW1iynywhXwMOwAhvGzCBClt3kmUAm235KTz1Glg888FK2PVzSY1u6qtSsTcDUUCJ3_o2LUICviukV1fUqd9t1EzP_V_4lO9Jx4CR",
                            height: 176, yOffset: 0, totalWidth: geometry.size.width)
                        phoneImage(
                            url:
                                "https://lh3.googleusercontent.com/aida-public/AB6AXuCg091i4f0WJO63g9R2MYlHnNsNj9ietAEt0alu3p_Gv7kC4JT_nSxqhB08EWp9OfAHdy7Rj0ruLnoXD-Rmr0Muiah_EoZme97POGFxEGjiqiRa3Blqk3gIhtA5VPFEP3ZPHyGxDbZOXUtZBATJcBrA5WG5tHRQND8hCBgx_07XkBH8mt4EFGDg6BlK3VOVh76jMkYj_lW5N-htmvzStxbHJxr1T3FEhfS97iw9UYThYNevDnjbBAAUYkl977CRmQtpRVAHTzkGSi7-",
                            height: 144, yOffset: 12, totalWidth: geometry.size.width)
                    }
                    .padding(.horizontal, 8)
                }
                .padding(.top, 32)
            }
            .frame(minHeight: 500)
            .cornerRadius(32)
            .padding(.horizontal, 16)
            .padding(.top, 8)
        }
        .frame(height: 520)  // Give fixed height for GeometryReader to work within ScrollView
    }

    private func phoneImage(url: String, height: CGFloat, yOffset: CGFloat, totalWidth: CGFloat)
        -> some View
    {
        AsyncImage(url: URL(string: url)) { image in
            image
                .resizable()
                .aspectRatio(contentMode: .fill)
        } placeholder: {
            Color.gray.opacity(0.3)
        }
        .frame(width: (totalWidth - 64) / 4, height: height)
        .cornerRadius(12, corners: [.topLeft, .topRight])
        .shadow(color: .black.opacity(0.2), radius: 10, y: 5)
        .offset(y: yOffset)
    }

    // MARK: - Shop by Brand
    private var shopByBrandSection: some View {
        VStack(alignment: .leading, spacing: 16) {
            HStack {
                Text("Shop by Brand")
                    .font(.system(size: 20, weight: .bold))
                    .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))

                Spacer()

                Button(action: {}) {
                    Text("See All")
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                }
            }

            HStack(spacing: 0) {
                brandItem(icon: "apple.logo", name: "Apple")
                Spacer()
                brandItem(icon: "smartphone", name: "Samsung")  // approximated icon
                Spacer()
                brandItem(icon: "g.circle.fill", name: "Google")  // approximated icon
                Spacer()
                brandItem(icon: "ellipsis.circle", name: "Others")  // approximated icon
            }
        }
        .padding(.horizontal, 24)
        .padding(.top, 32)
    }

    private func brandItem(icon: String, name: String) -> some View {
        VStack(spacing: 8) {
            ZStack {
                Circle()
                    .fill(Color.white)
                    .frame(width: 56, height: 56)
                    .shadow(color: .black.opacity(0.05), radius: 4, y: 2)

                Image(systemName: icon)
                    .font(.system(size: 24))
                    .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
            }

            Text(name)
                .font(.system(size: 12, weight: .medium))
                .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))  // Dark navy
        }
    }

    // MARK: - Trending Now
    private var trendingNowSection: some View {
        VStack(alignment: .leading, spacing: 16) {
            Text("Trending Now")
                .font(.system(size: 20, weight: .bold))
                .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                .padding(.horizontal, 24)

            // Trending Card
            ZStack(alignment: .trailing) {
                Color(red: 1.0, green: 0.92, blue: 0.92)  // #FFEBEB

                HStack {
                    VStack(alignment: .leading, spacing: 4) {
                        Text("NEW ARRIVAL")
                            .font(.system(size: 10, weight: .bold))
                            .tracking(2)
                            .foregroundColor(.red)
                            .padding(.bottom, 4)

                        Text("iPhone 15\nSeries")
                            .font(.system(size: 24, weight: .bold))
                            .foregroundColor(Color(red: 0.0, green: 0.11, blue: 0.27))
                            .lineLimit(2)

                        Text("Save up to 20% today")
                            .font(.system(size: 14))
                            .foregroundColor(Color.gray)
                            .padding(.bottom, 12)

                        Button(action: {}) {
                            Text("Explore")
                                .font(.system(size: 12, weight: .bold))
                                .foregroundColor(.white)
                                .padding(.horizontal, 16)
                                .padding(.vertical, 8)
                                .background(Color(red: 0.0, green: 0.11, blue: 0.27))
                                .cornerRadius(999)
                        }
                    }
                    .padding(.leading, 24)
                    .padding(.vertical, 24)

                    Spacer()

                    AsyncImage(
                        url: URL(
                            string:
                                "https://lh3.googleusercontent.com/aida-public/AB6AXuBVV4YQT7IkJity6Ll4xYYghfXwEIIkTuqN-tN8iFCCYUakCOEcQuw4-oimostQWNU6COhUpptI2X5yR4gsFmNKmN8HYJE5iGnMX0pOgHBKx7pxq7PrYjzbc9zK6FoXhCSp8f3sDye0IOeWnTK_bmys5g1watgJWkCIXOdrZMmY6d_JI_X4jzVYILVz0XFQOXUBBTMxbANODeO84DENHZ3mg6nFH0Ll7feupW-r1PKtePcnKh1ZhPOaSdy_h0yz5LdEyff6-IbvM7sO"
                        )
                    ) { image in
                        image
                            .resizable()
                            .aspectRatio(contentMode: .fit)
                    } placeholder: {
                        Color.clear
                    }
                    .frame(width: 128, height: 128)
                    .padding(.trailing, 16)
                }
            }
            .cornerRadius(24)
            .padding(.horizontal, 24)
        }
        .padding(.top, 40)
    }

    // MARK: - Bottom Nav
    private var bottomNavView: some View {
        VStack(spacing: 0) {
            HStack(spacing: 0) {
                navItem(icon: "house.fill", title: "Home", isActive: true)
                Spacer()
                navItem(icon: "square.grid.2x2", title: "Shop", isActive: false)
                Spacer()
                navItem(icon: "heart", title: "Wishlist", isActive: false)
                Spacer()
                navItem(icon: "person", title: "Profile", isActive: false)
            }
            .padding(.horizontal, 32)
            .padding(.vertical, 16)
            .background(Color.white)
            .overlay(
                Rectangle()
                    .frame(height: 1)
                    .foregroundColor(Color(red: 0.94, green: 0.95, blue: 0.96)),  // border-slate-100
                alignment: .top
            )

            // Home Indicator
            RoundedRectangle(cornerRadius: 3)
                .fill(Color(red: 0.8, green: 0.8, blue: 0.8))
                .frame(width: 128, height: 4)
                .padding(.bottom, 8)
                .padding(.top, 4)
        }
        .background(Color.white)
    }

    private func navItem(icon: String, title: String, isActive: Bool) -> some View {
        Button(action: {}) {
            VStack(spacing: 4) {
                Image(systemName: icon)
                    .font(.system(size: 24))
                Text(title)
                    .font(.system(size: 10, weight: .bold))
            }
            .foregroundColor(
                isActive
                    ? Color(red: 0.0, green: 0.11, blue: 0.27)
                    : Color(red: 0.58, green: 0.64, blue: 0.72))  // slate-400
        }
    }
}

struct GrandMobilesView_Previews: PreviewProvider {
    static var previews: some View {
        GrandMobilesView()
    }
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/pages/GrandMobilesView.kt
package com.ecommerceearn.app.ui.pages

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage

@Composable
fun GrandMobilesView(onNavigateBack: () -> Unit = {}) {
    val bgLight = Color(0xFFF8FAFC)
    
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(bgLight)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
        ) {
            GrandMobilesHeader(onNavigateBack)
            GrandMobilesHero()
            GrandMobilesShopByBrand()
            GrandMobilesTrendingNow()
            Spacer(modifier = Modifier.height(120.dp))
        }
        
        GrandMobilesBottomNav(modifier = Modifier.align(Alignment.BottomCenter))
    }
}

@Composable
private fun GrandMobilesHeader(onNavigateBack: () -> Unit) {
    val navyColor = Color(0xFF001B44)
    
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 24.dp, vertical = 16.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        IconButton(
            onClick = onNavigateBack,
            modifier = Modifier.background(Color.White.copy(alpha = 0.2f), CircleShape).size(40.dp)
        ) {
            Icon(Icons.Default.ArrowBack, contentDescription = "Back", tint = navyColor)
        }
        
        Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
            IconButton(
                onClick = {},
                modifier = Modifier.background(Color.White.copy(alpha = 0.2f), CircleShape).size(40.dp)
            ) {
                Icon(Icons.Default.Search, contentDescription = "Search", tint = navyColor)
            }
            IconButton(
                onClick = {},
                modifier = Modifier.background(Color.White.copy(alpha = 0.2f), CircleShape).size(40.dp)
            ) {
                Icon(Icons.Default.ShoppingCart, contentDescription = "Cart", tint = navyColor)
            }
        }
    }
}

@Composable
private fun GrandMobilesHero() {
    val navyColor = Color(0xFF001B44)
    
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(520.dp)
            .padding(horizontal = 16.dp, vertical = 8.dp)
            .clip(RoundedCornerShape(32.dp))
            .background(
                Brush.linearGradient(
                    colors = listOf(
                        Color(0xFF4FB0AE),
                        Color(0xFFD4E2D4),
                        Color(0xFFFFD670),
                        Color(0xFFFF9248)
                    )
                )
            )
    ) {
        // Overlay mimic
        Box(modifier = Modifier.fillMaxSize().background(Color.Black.copy(alpha = 0.05f)))
        
        // Sparkles (positioned manually)
        Icon(Icons.Default.Star, contentDescription = null, tint = navyColor, modifier = Modifier.offset(x = 40.dp, y = 48.dp).size(24.dp))
        Icon(Icons.Default.Star, contentDescription = null, tint = navyColor.copy(alpha = 0.4f), modifier = Modifier.align(Alignment.TopEnd).offset(x = (-48).dp, y = 128.dp).size(24.dp))
        Icon(Icons.Default.Star, contentDescription = null, tint = navyColor, modifier = Modifier.align(Alignment.BottomStart).offset(x = 64.dp, y = (-160).dp).size(24.dp))

        Column(
            modifier = Modifier.fillMaxSize().padding(top = 32.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Text("Betul's Exclusive", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = navyColor, letterSpacing = (-0.5).sp, modifier = Modifier.padding(bottom = 8.dp))
            Text("GRAND MOBILES\nSALE", fontSize = 42.sp, fontWeight = FontWeight.Black, color = navyColor, textAlign = androidx.compose.ui.text.style.TextAlign.Center, lineHeight = 38.sp, modifier = Modifier.padding(bottom = 32.dp))
            
            Button(
                onClick = {},
                colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent),
                contentPadding = PaddingValues(horizontal = 40.dp, vertical = 14.dp),
                modifier = Modifier.border(2.dp, navyColor, RoundedCornerShape(0.dp)).padding(bottom = 48.dp)
            ) {
                Text("GRAB NOW", fontSize = 18.sp, fontWeight = FontWeight.Black, letterSpacing = 2.sp, color = navyColor)
            }
            
            Spacer(modifier = Modifier.weight(1f))

            Row(
                modifier = Modifier.fillMaxWidth().padding(horizontal = 8.dp),
                verticalAlignment = Alignment.Bottom,
                horizontalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                GrandMobilesPhoneImage(
                    url = "https://lh3.googleusercontent.com/aida-public/AB6AXuAxDDaHdTrwru2LB1m_bJMDHJJXXNMH6lz9-9S8rddKmEqEWunUOUFNIbDEZPU2d-YJNwhSmwIpL5fO0epqwuAxkKHsryedI3kKY1upVgvUY8zIeQO-cDyHSq6-D05NQA0G6nx_pJudqZF6b3iWA1wThCbVLqJxo19m7Zr0YUMv3dwjt7Sy6J4CKABFbQYaor4Ku67YWhfhByHnwYFOTfjsu2ML2U_ovNFWJfYxos7yBMXhlZSbcKqy41LlXGZAyT3a7GN-2zDoKY5S",
                    height = 128, yOffset = 16, modifier = Modifier.weight(1f)
                )
                GrandMobilesPhoneImage(
                    url = "https://lh3.googleusercontent.com/aida-public/AB6AXuCksjqMHl_Le2OSvtMgyqEeMo6F3kf3fFLKon7RNfxaxt4-VznnUV60f8NxHfim6HzcFbPp8g1xan7gXkkXG7QTymUM5a914tPRdFunL5GAdNouDtxDqqI6FpYeDg6JV-k3vMn3Vv59V5b21elRNYMVaMKilZnTQRoZyFd8qQTod4AQNKx_50Hsy9Guqv8as_l56QtYgg078rgL15j1EsNxw-sR6S-PfuyZuHLEfjxyMvpmJXWCuVnkuVWnVFPiHYLR7HYRSG6lytlr",
                    height = 160, yOffset = 8, modifier = Modifier.weight(1f)
                )
                GrandMobilesPhoneImage(
                    url = "https://lh3.googleusercontent.com/aida-public/AB6AXuC4DwQQFVyxxqGSPaN4wzRr5wT8_Pxg_2e8k2v_Yl0MZKth5yEZ0-GNsJqwmF9RPLRiCD-rjf7we_aMQjZpHUtp1TvWjCLfelyn9fmQDv_cr3EY2QCHDdBw5TOaVRYb7CGmoppynojkVeRqDBe1jHlv9sYjOsyksxwT0JUxoEZLW1iynywhXwMOwAhvGzCBClt3kmUAm235KTz1Glg888FK2PVzSY1u6qtSsTcDUUCJ3_o2LUICviukV1fUqd9t1EzP_V_4lO9Jx4CR",
                    height = 176, yOffset = 0, modifier = Modifier.weight(1f)
                )
                GrandMobilesPhoneImage(
                    url = "https://lh3.googleusercontent.com/aida-public/AB6AXuCg091i4f0WJO63g9R2MYlHnNsNj9ietAEt0alu3p_Gv7kC4JT_nSxqhB08EWp9OfAHdy7Rj0ruLnoXD-Rmr0Muiah_EoZme97POGFxEGjiqiRa3Blqk3gIhtA5VPFEP3ZPHyGxDbZOXUtZBATJcBrA5WG5tHRQND8hCBgx_07XkBH8mt4EFGDg6BlK3VOVh76jMkYj_lW5N-htmvzStxbHJxr1T3FEhfS97iw9UYThYNevDnjbBAAUYkl977CRmQtpRVAHTzkGSi7-",
                    height = 144, yOffset = 12, modifier = Modifier.weight(1f)
                )
            }
        }
    }
}

@Composable
private fun GrandMobilesPhoneImage(url: String, height: Int, yOffset: Int, modifier: Modifier) {
    AsyncImage(
        model = url,
        contentDescription = null,
        contentScale = ContentScale.Crop,
        modifier = modifier
            .offset(y = yOffset.dp)
            .height(height.dp)
            .clip(RoundedCornerShape(topStart = 12.dp, topEnd = 12.dp))
            .background(Color.Gray.copy(alpha = 0.3f))
    )
}

@Composable
private fun GrandMobilesShopByBrand() {
    val navyColor = Color(0xFF001B44)
    
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 24.dp)
            .padding(top = 32.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
            Text("Shop by Brand", fontSize = 20.sp, fontWeight = FontWeight.Bold, color = navyColor)
            Text("See All", fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = navyColor)
        }
        
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            GrandMobilesBrandItem(Icons.Default.PhoneIphone, "Apple")
            GrandMobilesBrandItem(Icons.Default.Smartphone, "Samsung")
            GrandMobilesBrandItem(Icons.Default.AccountCircle, "Google")
            GrandMobilesBrandItem(Icons.Default.MoreHoriz, "Others")
        }
    }
}

@Composable
private fun GrandMobilesBrandItem(icon: ImageVector, name: String) {
    val navyColor = Color(0xFF001B44)
    Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Box(
            modifier = Modifier
                .size(56.dp)
                .shadow(4.dp, CircleShape, spotColor = Color.Black.copy(alpha = 0.05f))
                .background(Color.White, CircleShape),
            contentAlignment = Alignment.Center
        ) {
            Icon(icon, contentDescription = name, tint = navyColor, modifier = Modifier.size(24.dp))
        }
        Text(name, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = navyColor)
    }
}

@Composable
private fun GrandMobilesTrendingNow() {
    val navyColor = Color(0xFF001B44)
    
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(top = 40.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Text("Trending Now", fontSize = 20.sp, fontWeight = FontWeight.Bold, color = navyColor, modifier = Modifier.padding(horizontal = 24.dp))
        
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp)
                .clip(RoundedCornerShape(24.dp))
                .background(Color(0xFFFFEBEB)) // #FFEBEB
        ) {
            Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                Column(modifier = Modifier.padding(start = 24.dp, top = 24.dp, bottom = 24.dp), verticalArrangement = Arrangement.spacedBy(4.dp)) {
                    Text("NEW ARRIVAL", fontSize = 10.sp, fontWeight = FontWeight.Bold, letterSpacing = 2.sp, color = Color.Red, modifier = Modifier.padding(bottom = 4.dp))
                    Text("iPhone 15\nSeries", fontSize = 24.sp, fontWeight = FontWeight.Bold, color = navyColor, lineHeight = 28.sp)
                    Text("Save up to 20% today", fontSize = 14.sp, color = Color.Gray, modifier = Modifier.padding(bottom = 12.dp))
                    
                    Button(
                        onClick = {},
                        shape = CircleShape,
                        colors = ButtonDefaults.buttonColors(containerColor = navyColor),
                        contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp),
                        modifier = Modifier.height(32.dp)
                    ) {
                        Text("Explore", fontSize = 12.sp, fontWeight = FontWeight.Bold, color = Color.White)
                    }
                }
                
                AsyncImage(
                    model = "https://lh3.googleusercontent.com/aida-public/AB6AXuBVV4YQT7IkJity6Ll4xYYghfXwEIIkTuqN-tN8iFCCYUakCOEcQuw4-oimostQWNU6COhUpptI2X5yR4gsFmNKmN8HYJE5iGnMX0pOgHBKx7pxq7PrYjzbc9zK6FoXhCSp8f3sDye0IOeWnTK_bmys5g1watgJWkCIXOdrZMmY6d_JI_X4jzVYILVz0XFQOXUBBTMxbANODeO84DENHZ3mg6nFH0Ll7feupW-r1PKtePcnKh1ZhPOaSdy_h0yz5LdEyff6-IbvM7sO",
                    contentDescription = null,
                    modifier = Modifier.size(128.dp).padding(end = 16.dp),
                    contentScale = ContentScale.Fit
                )
            }
        }
    }
}

@Composable
private fun GrandMobilesBottomNav(modifier: Modifier = Modifier) {
    val navyColor = Color(0xFF001B44)
    val inactiveColor = Color(0xFF94A3B8) // slate-400 equivalent
    
    Column(
        modifier = modifier
            .fillMaxWidth()
            .background(Color.White)
    ) {
        Box(modifier = Modifier.fillMaxWidth().height(1.dp).background(Color(0xFFF1F5F9)))
        Row(
            modifier = Modifier.fillMaxWidth().padding(horizontal = 32.dp, vertical = 16.dp),
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            GrandMobilesNavItem(Icons.Default.Home, "Home", true, navyColor, inactiveColor)
            GrandMobilesNavItem(Icons.Default.Dashboard, "Shop", false, navyColor, inactiveColor)
            GrandMobilesNavItem(Icons.Default.Favorite, "Wishlist", false, navyColor, inactiveColor)
            GrandMobilesNavItem(Icons.Default.Person, "Profile", false, navyColor, inactiveColor)
        }
        
        Box(
            modifier = Modifier.fillMaxWidth().padding(top = 4.dp, bottom = 8.dp),
            contentAlignment = Alignment.Center
        ) {
            Box(modifier = Modifier.width(128.dp).height(4.dp).background(Color(0xFFCCCCCC), RoundedCornerShape(3.dp)))
        }
    }
}

@Composable
private fun GrandMobilesNavItem(icon: ImageVector, title: String, isActive: Boolean, activeColor: Color, inactiveColor: Color) {
    val color = if (isActive) activeColor else inactiveColor
    Column(horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(4.dp)) {
        Icon(icon, contentDescription = title, tint = color, modifier = Modifier.size(24.dp))
        Text(title, fontSize = 10.sp, fontWeight = FontWeight.Bold, color = color)
    }
}
